src/Illuminate/Database/Eloquent/Concerns/HasCustomizations.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Illuminate\Database\Eloquent\Concerns;
+
+use Illuminate\Database\Eloquent\Attributes\PrimaryKey;
+use ReflectionClass;
+
+trait HasCustomizations
+{
+    /**
+     * Apply customizations from attributes.
+     *
+     * @return void
+     */
+    protected function applyCustomizations()
+    {
+        $primaryKeyAttribute = $this->resolveCustomPrimaryKey();
+        if (! $primaryKeyAttribute) {
+            return;
+        }
+
+        $this->setKeyName($primaryKeyAttribute->name);
+        $this->setKeyType($primaryKeyAttribute->type);
+        $this->setIncrementing($primaryKeyAttribute->incrementing);
+    }
+
+    /**
+     * Resolve the custom primary key from the attributes.
+     *
+     * @return PrimaryKey|null
+     */
+    protected function resolveCustomPrimaryKey()
+    {
+        $reflectionClass = new ReflectionClass(static::class);
+        /** @var $primaryKeyAttribute \ReflectionAttribute|null */
+        $primaryKeyAttribute = collect($reflectionClass->getAttributes(PrimaryKey::class))
```

Comment:
Do we really need to use collection here, especially the global helper method?

Reply:
TBH, IDK why I used it, now that you mention it is totally unnecessary, updating now, thanks!